Logan/guide enhancements

loganallen opened the pull request:
Addresses issue #63

eappel commented on Classes/Eat Now/Root View Controller/EatNowTableViewController.swift:
```diff
@@ -334,10 +334,8 @@ class EatNowTableViewController: UITableViewController, UISearchResultsUpdating,
-                var stringMatch = eatery.name.lowercaseString.rangeOfString(searchText.lowercaseString)
-                return (stringMatch != nil)
-            })
+            let pred = NSPredicate(format: "name contains[cd] %@ OR menu.description contains[cd] %@", searchText, searchText)
```
Are we guaranteed that ```eatery.menu != nil``` here?

alexzielenski replied:
Yes, I added a line into Eatery.swift to initialize the menu beforehand. Even if it was nil, menu.description would just be (null) so it's not really a big issue.

eappel replied:
Loading the menu is an async network request so we aren't guaranteed it will finish before a user searches.  But if it being nil won't crash the search then we good :+1:

alexzielenski commented:
removing Keys.plist is in its own commit. I think you mean pull request, and you're probably right but the commit history of master after merge will still have it in its own commit after merging.

eappel commented on Classes/Data/Models/Eatery.swift:
```diff
@@ -53,9 +53,17 @@ class Eatery: NSObject {
         self.calendar = MXLCalendar()
+        
+        super.init()
+        loadTodaysMenu(true, completion: {})
```
Placing the menu load here is interesting.  Since the eateries get initialized on app launch, the menus will now start loading before the hours.  The hours are the higher priority data since we need it to display the feed.  Will also loading the menus on launch delay the hours request?

alexzielenski commented:
I don't see how using NSArray is an "anti-pattern" since that is the underlying Foundation class for arrays even in Swift so I'm just using API.  Using NSPredicate is faster to write and cleaner IMO.

eappel commented:
Removing the Keys.plist is in f68e8a708aee0b499660d6d1e6db26e13bda0ca1...

alexzielenski commented:
Oh right so what was the issue you had with it?

alexzielenski commented:
commit has been renamed, too

eappel commented:
Just separate it into its own commit before merging.  Don't push remote after rebasing though because it will nuke the conversation history.  Treat as append-only before merging

eappel commented:
NSArray is anti-pattern in because it isn't explicitly typed

eappel commented:
Otherwise looks good :ship:

alexzielenski commented:
merged

eappel commented:
Next time link the merge SHA in the "merge" comment :)

ChenJesse committed:
Alternate dining hall names added for any names that arent covered by official name or nickname

anniexcheng committed:
Add look ahead files

Show all dining halls and change hour colors

Make info icon on eatery guide go to detailVC for selected date

Make DetailVC display hours and menu for selected meal

Update look ahead menus according to new design

Add carrot for expanded eatery cells on lookAheadVC

Add more spacing between menu items

Add date to detailVC navbar title

Make miscellaneous UI changes and code cleanup

Add look ahead files

Show all dining halls and change hour colors

Make info icon on eatery guide go to detailVC for selected date

Make DetailVC display hours and menu for selected meal

Update look ahead menus according to new design

Add carrot for expanded eatery cells on lookAheadVC

Add more spacing between menu items

Add date to detailVC navbar title

Make DetailVC display hours and menu for selected meal

Add carrot for expanded eatery cells on lookAheadVC

Add more spacing between menu items

Make miscellaneous UI changes and code cleanup

dantheli committed:
Change refresh behavior, resolves #111

Add manual refresh menu item.

lucasderraugh committed:
Resolve merge conflicts

anniexcheng committed:
Fix minor changes caused by rebasing

ChenJesse committed:
Final changes

say25 committed:
Nickname

Add Sage to Atrium Cafe
Handle ‘ problems

dantheli committed:
Clean up and add comments

anniexcheng committed:
Create private instance of TitleDateFormatter for reuse in MenuViewController

anniexcheng committed:
Replace MiscellaneousExtensions with String+Manipulation to follow Obj-C extension syntax

anniexcheng committed:
Add private instance of NSDateFormatter in LookAheadVC

anniexcheng committed:
Add sorting for eateries

anniexcheng committed:
Change Team to None

anniexcheng committed:
Remove Eatery object from table view cells for MVC separation

dantheli committed:
Rebase with master

Change reload behavior

anniexcheng committed:
Merge pull request #10 from cuappdev/annie/look-ahead-rebased

Annie/look ahead rebased

ChenJesse committed:
Searchbar repositioned, cancel button clears search

dantheli committed:
Merge pull request #5 from cuappdev/daniel/persistence

Implement Eatery Data Persistence

ChenJesse committed:
Merge branch 'jesse/searchbar-reposition'

anniexcheng committed:
Change Eatery grid cells according to design and make font changes

mlelien committed:
changed sorting values so it's more a part of Eatery.extensions

lucasderraugh committed:
Merge branch 'emily/refactor-eatery-sorting-values'

lucasderraugh committed:
Remove C for loop for Swift 3.0

anniexcheng committed:
Make miscellaneous UI changes

lucasderraugh committed:
iPhone 5 or less will use 1 across layout, else 2 across
Closes #2

dantheli committed:
Remove disk reading and empty closure

mlelien committed:
scroll to lunch for lite lunch

lucasderraugh committed:
Merge pull request #14 from cuappdev/daniel/remove-persistence

Revert to old fetch functionality

anniexcheng committed:
Make minor code style change

anniexcheng committed:
Fix menu sharing on lookAheadVC and change sort icon

anniexcheng committed:
Merge branch 'annie/design-polish'

anniexcheng committed:
Change sort icon

anniexcheng committed:
Remove status bar from launch screen

mlelien committed:
removed lite lunch

Commit:
Fix delegate assignment for MenuViewController

dantheli committed:
Fix toggle menu button

dantheli committed:
Expand search to include changing menus

dennisfedorko committed:
Remove funnel button

anniexcheng committed:
Merge pull request #21 from cuappdev/emily/remove-empty-lite-lunch

removed lite lunch, close #7

anniexcheng committed:
Merge pull request #20 from cuappdev/daniel/fix-toggle-menu-button

Fix toggle menu button, close #16

anniexcheng committed:
Merge pull request #19 from cuappdev/mark/fix-favorites-from-look-ahead

Fix Marking Favorites from Look Ahead, close #17

anniexcheng committed:
Fix look ahead feature to display proper times

anniexcheng committed:
Merge pull request #24 from cuappdev/annie/look-ahead-fix

Fix look ahead feature to display proper times, close #23

anniexcheng committed:
Move initial collection view content offset down a little from navbar

anniexcheng committed:
Change eatery name on detailVC to nickname

anniexcheng committed:
Make app look good on iPhone 4 and 5 screens

anniexcheng committed:
Change Jansen's Dining to Bethe House Dining

dantheli committed:
Conform to new design, remove prints, remove status label

dantheli committed:
Change closed color to dark

lucasderraugh committed:
Add Natasha's sorting menus

lucasderraugh committed:
Add lazy evaluation of menu items
Close #22

lucasderraugh committed:
Merge branch 'daniel/search-changing-menus'

lucasderraugh committed:
Merge annie/iphone-4-5-compatibility into master

lucasderraugh committed:
Merge daniel/watch-polish into master

lucasderraugh committed:
Change bundle identifier

lucasderraugh committed:
Change from com to org

lucasderraugh committed:
Change today extension identifier

kevingreer committed:
quick actions to favorites

kevingreer committed:
removed whitespace

kevingreer committed:
removed unneccesary self's

kevingreer committed:
Merge branch 'kevin/FTQuickActions'

Commit:
search view stays visible while sorting and searching

Commit:
adding collection view now in viewDidLoad

say25 committed:
Host Images Elsewhere

http://imgur.com/a/47SfN

say25 committed:
Remove images from repo, upload Icon Image as well

say25 committed:
Remove spaces, Info Button Hides

Addresses #39

anniexcheng committed:
Merge pull request #40 from cuappdev/steven/hide-info-button

Remove spaces, Info Button Hides, Close #39

lucasderraugh committed:
Change watch assets to have no alpha channel

lucasderraugh committed:
Don't force fetch on sort

lucasderraugh committed:
Remove extra variable for tracking active search

lucasderraugh committed:
Merge branch 'natasha/searchAndSort'

Commit:
collection view has 3 cells per row on iPad

Commit:
universal

say25 committed:
Merge pull request #42 from cuappdev/natasha/universal

collection view has 3 cells per row on iPad

Commit:
Alphabetical and Open/Alphabetical sorting for watch app

Commit:
Deleted unnecessary variable

Commit:
Watch app initially sorts by open and alphabetical

Close #32

say25 committed:
Fix Opens/Closes in 0 minutes

Example: Mattins closes at 10pm. At 9:59pm it would display closing in
0m rather than closing in 1m

ChenJesse committed:
Implement NSOperationQueue for sorting, reload only the sections

ChenJesse committed:
Merge branch 'jesse/keyboard-response-time'

say25 committed:
Add watch icons and add all images to one album

lucasderraugh committed:
Make a little Swiftier

lucasderraugh committed:
Merge logan/watch_app_sorting into master

lucasderraugh committed:
Swiftier code, cleaning up as one does

say25 committed:
Add Release Notes to README

ChenJesse committed:
Preserves sorting option

ChenJesse committed:
Merge branch 'jesse/preserve-sorting'

Commit:
Changed guide cells

Commit:
Address issue #63

lucasderraugh commented on Classes/Eat Now/Detail View Controller/TableView Cells/EateryHeaderTableViewCell.swift:
```diff
@@ -27,7 +26,7 @@ class EateryHeaderTableViewCell: UITableViewCell {
         super.awakeFromNib()
         selectionStyle = .None
-        infoButton.tintColor = UIColor(red: 0.0, green: 0.0, blue: 0.0, alpha: 0.5)
+//        infoButton.tintColor = UIColor(red: 0.0, green: 0.0, blue: 0.0, alpha: 0.5)
```
No commented out code please

Commit:
Removed commented code
